### PR TITLE
Add basic debian package metadata

### PR DESCRIPTION
**Pull request by adrianheine**
See #14.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -87,4 +87,38 @@ gitsh is packaged and installed using GNU autotools.
 
         make release
 
+## Building the dpkg package
+
+You first need to set up a working directory for your build with a copy of
+gitsh (preferably obtained via git):
+
+  - mkdir build_gitsh && cd build_gitsh/
+  - git clone https://github.com/thoughtbot/gitsh.git gitsh-0.6
```
Comment (adrianheine) [inline CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -87,4 +87,38 @@ gitsh is packaged and installed using GNU autotools.
...
+
+  - sudo aptitude install devscripts dh-autoreconf ruby-rspec ruby-bourne pry
+  - debuild -us -uc
+
+There are some common errors you might get when running `debuild`:
+
+  - Vendorizing gems failed
+    - Just re-run debuild
+  - Aborting due to unexpected upstream changes (often config.log)
+    - Delete config.log or update the orig tarball
```
Comment (adrianheine) [inline CONTRIBUTING.md]: Done.

**Comment by adrianheine** [comment]
I merged with current master and addressed the two comments.

**Comment by mike-burns** [comment]
Code looks good. I'll try this on Friday.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -87,4 +95,39 @@ gitsh is packaged and installed using GNU autotools.
...
+    cd ..
+    mv gitsh gitsh-0.10/
+
+or
+
+    ln -s gitsh gitsh-0.10
+
+Next, create the orig tarball:
+
+    tar hczf gitsh_0.10.orig.tar.gz gitsh-0.10/
```
Comment (mike-burns) [inline CONTRIBUTING.md]: I'm surprised that this isn't a `make dist` instead.

File: debian/control [hunk 4]
```diff
@@ -0,0 +1,17 @@
+Source: gitsh
+Maintainer: thoughtbot <support@thoughtbot.com>
+Section: misc
+Priority: optional
+Standards-Version: 3.9.5
+Build-Depends: debhelper (>= 9), dh-autoreconf, pry, ruby2.2, ruby-bourne,
```
Comment (mike-burns) [inline debian/control]: There is no `ruby2.2` in `stable`, but there is `ruby2.1`. Is it possible to relax this requirement to "anything compatible with Ruby 2"?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -87,4 +95,39 @@ gitsh is packaged and installed using GNU autotools.
...
+    cd gitsh-0.10/
+
+Now you need to install the build-dependencies and then run the build process:
+
+    sudo aptitude install devscripts dh-autoreconf ruby-rspec ruby-bourne pry
```
Comment (mike-burns) [inline CONTRIBUTING.md]: I also needed `ruby-parselet`.

File: debian/rules [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/make -f
+%:
+	RUBY=`which ruby2.2` dh $@ --with autoreconf
```
Comment (mike-burns) [inline debian/rules]: I got this to work with this addition:

```make
override_dh_autoreconf:
	dh_autoreconf ./autogen.sh
```

**Comment by mike-burns** [comment]
I needed to modify `autogen.sh` to add `-c` to `automake`:

```sh
#!/bin/sh

aclocal -I m4 && autoconf && automake --add-missing -c
```

And I removed the existing symlinks (in zsh: `rm -f **/*(@)`).

---

In `spec/Makefile.am`, the `RB_LOG_COMPILER` is `rspec`. This works fine when using a chruby, but was unpredictable when using system Ruby. Changing it to `bundle exec rspec` fixed problems for me. @georgebrock should this be done in general?

File: debian/control (hunk 4 above)
Comment (adrianheine) [inline debian/control]: Done.

File: CONTRIBUTING.md (hunk 5 above)
Comment (adrianheine) [inline CONTRIBUTING.md]: Right, `ruby-parslet`.

File: CONTRIBUTING.md (hunk 3 above)
Comment (adrianheine) [inline CONTRIBUTING.md]: That would need `automake` and `./configure` runs upfront.

File: debian/rules (hunk 6 above)
Comment (adrianheine) [inline debian/rules]: What issue did you ran into with using `autoreconf`?

Comment (mike-burns) [inline debian/rules]: Oh interesting: it works now. Nevermind!

**Comment by georgebrock** [comment]
Closing in favour of https://github.com/thoughtbot/apt/pull/1

**Pull request closed without merging**